src/sage/rings/polynomial/multi_polynomial_libsingular.pyx

```diff
@@ -5757,6 +5757,83 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
         else:
             return self * self.denominator()
 
+    def in_subalgebra(self, J, algorithm="algebra_containment"):
+        """
+        Return whether this polynomial is contained in the subalgebra
+        generated by ``J``
+
+        INPUT:
+
+        - ``J`` -- list of elements of the parent polynomial ring
+
+        - ``algorithm`` -- can be "algebra_containment" (the default),
+          "inSubring", or "groebner".
+
+          - "algebra_containment" (default): use Singular's
+            ``algebra_containment`` function,
+            https://www.singular.uni-kl.de/Manual/4-2-1/sing_1247.htm#SEC1328. The
+            Singular documentation suggests that this is frequently
+            faster than the next option.
+
+          - "inSubring": use Singular's ``inSubring`` function,
+            https://www.singular.uni-kl.de/Manual/4-2-0/sing_1240.htm#SEC1321.
+
+          - "groebner": use the algorithm described in Singular's
+            documentation, but within Sage: if the subalgebra
+            generators are `y_1`, ..., `y_m`, then create a new
+            polynomial algebra with the old generators along with new
+            ones: `z_1`, ..., `z_m`. Create the ideal `(z_1 - y_1,
+            ..., z_m - y_m)`, and reduce the polynomial modulo this
+            ideal. The polynomial is contained in the subalgebra if
+            and only if the remainder involves only the new variables
+            `z_i`.
+
```

Comment:
Matthias is massively correcting string literals as in this:
```diff
    def set_random_number_generator(self, rng = 'default'):
        """
-       Set the gsl random number generator to be one of ``default``,
-       ``luxury``, or ``taus``.
+       Set the gsl random number generator to be one of ``'default'``,
+       ``'luxury'``, or ``'taus'``.

        EXAMPLES::
```
So how about changing `"groebner"` (and others) to 
```
``'groebner'`` 
```
? This would save him from some work.

Reply:
Done. The built documentation looks okay to me.